sklearn/inspection/_metric_threshold_curve.py

```diff
@@ -0,0 +1,177 @@
+"""Metrics per threshold curves are used to assess performance on binary
+classification task given threshold grid. One can undestand the behaviour of
+threshold-dependent metrics when changing the threshold.
+"""
+
+# Authors: ########
+# License: BSD 3 clause
+
+from numbers import Integral, Real
+
+import numpy as np
+
+from ..utils import assert_all_finite, check_consistent_length, column_or_1d
+from ..utils._param_validation import Interval, validate_params
+from ..utils.multiclass import type_of_target
+from ..utils.validation import (
+    _check_pos_label_consistency,
+    _check_sample_weight,
+)
+
+
+@validate_params(
+    {
+        "y_true": ["array-like"],
+        "y_score": ["array-like"],
+        "score_func": [callable],
+        "threshold_grid": [
+            Interval(Integral, 3, None, closed="left"),
+            "array-like",
+            None,
+        ],
+        "pos_label": [Real, str, "boolean", None],
+        "sample_weight": ["array-like", None],
+    },
+    prefer_skip_nested_validation=True,
+)
+def metric_threshold_curve(
+    y_true,
+    y_score,
+    score_func,
```

Comment:
We might want to call this `scoring` as well for consistency. But here, we should only accept a callable.